### DISCUSSION TITLE
Question about using Unity

### DISCUSSION THREAD

**Discussion by arumons** [codemagic-ci-cd/codemagic-docs#2888] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Unity

### Steps to reproduce

I started using the Free plan for Unity development, but it seems that by default, no Unity version is installed on the machine.
(the directory under /Applications/Unity/Hub/Editor/ was empty.)

- Do I need to install Unity each time by referring to [this guide](https://docs.codemagic.io/knowledge-others/install-unity-version/) to use it?
- Is it possible to have Unity pre-installed on the machine to speed up the process?

### Expected results

none

### Actual results

none

**Comment by dam-ease** [comment, marked as answer]
Hello @arumons,
Yes, you have to add a specified version to be installed. I'm afraid installing can't be skipped or pre-installing on build machine such that it can be skipped isn't possible at the moment due to Unity requirements to run the Unity activate license and other build commands. There are available unity versions but they still need to be installed, if your desired Unity version isn't on the list, you can also contact us.

**Reply by arumons** [reply]
@dam-ease 
Thank you for your response.

I would be happy if Unity 6 (6000.x.xx) becomes available.

**Reply by dam-ease** [reply]
Hey @arumons
Glad to help.
Can you share the specific version of [Unity 6](https://unity.com/releases/editor/archive) you want to make use of here as well?

**Reply by arumons** [reply]
Thank you.
It would be helpful if I could use the latest version of Unity 6 currently released, which is 6000.0.31f.

**Reply by dam-ease** [reply]
Hey @arumons,
This will be available for a quick install next year but if you'd like to make use of this version before then, you can add below to your YAML to install it via custom scripts:

```
- name: Install Unity version
        script: |  
          yes Y | /Applications/Unity\ Hub.app/Contents/MacOS/Unity\ Hub -- --headless install --version $UNITY_VERSION --changeset $UNITY_VERSION_CHANGESET -a arm64 # -a x86_64 --y
          yes Y | /Applications/Unity\ Hub.app/Contents/MacOS/Unity\ Hub -- --headless install-modules --version $UNITY_VERSION -m ios android -a arm64 # -a x86_64 --y
```
Then set the `$UNITY_VERSION` and `$UNITY_VERSION_CHANGESET` variables in your environment variables.
<img width="1174" alt="Screenshot 2024-12-20 at 09 18 37" src="https://github.com/user-attachments/assets/de84e443-3062-4c6d-988e-dded4fdace01" />
